modules/consensus/accept.go

```diff
@@ -137,29 +137,20 @@ func (cs *ConsensusSet) validateHeader(tx dbTx, h types.BlockHeader) error {
 // tip. An error will be returned if block verification fails or if the block
 // does not extend the longest fork.
 //
-// addBlockToTree must use its own database update because it might need to
-// modify the database while returning an error on the block. To prevent error
-// tracking complexity, the error is handled inside the function so that 'nil'
-// can be appropriately returned by the database and the transaction can be
-// committed. Switching to a managed tx through bolt will make this complexity
+// addBlockToTree might need to modify the database while returning an error
+// on the block. Such errors are handled outside of the transaction by the
+// caller. Switching to a managed tx through bolt will make this complexity
 // unneeded.
-func (cs *ConsensusSet) addBlockToTree(b types.Block) (ce changeEntry, err error) {
-	var nonExtending bool
-	err = cs.db.Update(func(tx *bolt.Tx) error {
+func (cs *ConsensusSet) addBlockToTree(tx *bolt.Tx, b types.Block) (ce changeEntry, err error) {
+	err = func() error {
```

Comment:
I don't think this func needs to be here anymore, the error catching has been cleaned up some already, and since we've don't that I don't really see a reason to leave this func.